recipe/meta.yaml

```diff
@@ -21,7 +21,7 @@ requirements:
     - python                                 # [build_platform != target_platform]
     - cross-python_{{ target_platform }}     # [build_platform != target_platform]
     - numpy                                  # [build_platform != target_platform]
-    - pybind11                               # [build_platform != target_platform]
+    - pybind11 >=2.12.0,<3                   # [build_platform != target_platform]
```

Comment:
Also applied the same constraint in `requirements/build` as it was unconstrained previously